mubench.reviewsite/src/Controllers/ReviewsController.php

```diff
@@ -131,7 +139,55 @@ public function updateOrCreateReview($misuse_id, $reviewer_id, $comment, $findin
             $findingReview = FindingReview::firstOrNew(['review_id' => $review->id, 'rank' => $rank]);
             $findingReview->decision = $findings_review['hit'];
             $findingReview->save();
-            $findingReview->violation_types()->sync($findings_review['types']);
+            if($findings_review['types']){
+                $findingReview->violation_types()->sync($findings_review['types']);
+            }
+        }
+    }
+
+    private function collectAllMisuses($runs)
+    {
+        $all_misuses = new Collection;
+        foreach ($runs as $run) {
+            $all_misuses = $all_misuses->merge($run->misuses);
+        }
+        return $all_misuses;
+    }
+
+    private function determineNavigationTargets(Collection $all_misuses, $project_muid, $version_muid, $misuse_muid, $reviewer)
```

Comment:
I took the liberty to pull this into a method. Having all this additional logic in getReview certainly made that method to long. Please remember to do such cleanup.